api/buffer/design.go

```diff
@@ -526,6 +539,50 @@ var ImportConditions = Type("Conditions", func() {
 	Example(ExampleConditions())
 })
 
+// Task --------------------------------------------------------------------------------------------------------------
+
+var TaskID = Type("TaskID", String, func() {
+	Meta("struct:field:type", "= key.TaskID", "github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/key")
+	Description("Unique ID of the task.")
+	Example("task_1234")
+})
+
+var Task = Type("Task", func() {
+	Description("An asynchronous task.")
+	Attribute("id", TaskID)
+	Attribute("receiverId", ReceiverID)
+	Attribute("url", String, func() {
+		Description("URL of the task.")
+	})
+	Attribute("type", String)
+	Attribute("createdAt", String, func() {
+		Description("Date and time of the task creation.")
+		Format(FormatDateTime)
+		Example("2022-04-28T14:20:04+00:00")
+	})
+	Attribute("finishedAt", String, func() {
+		Description("Date and time of the task end.")
+		Format(FormatDateTime)
+		Example("2022-04-28T14:20:04+00:00")
+	})
+	Attribute("isFinished", Boolean)
+	Attribute("duration", Int64, func() {
+		Description("Duration of the task in milliseconds.")
+		Example(123456789)
+	})
+	Attribute("result", String)
+	Attribute("error", String)
+	Required("id", "receiverId", "url", "type", "createdAt", "finishedAt", "isFinished", "duration", "result", "error")
```

Comment:
Bolo by dobre pridat do testu nejaky nedokonceny task, `finishedAt=null,duration=0`, tu to mas medzi `required`.

Reply:
Neviem, ci by nebolo lepsie `finishedAt: null`, ako `finishedAt: <empty string>`.

Reply:
https://github.com/keboola/keboola-as-code/pull/1190/commits/b63c34def1923018cafc8fe6b190a7a652e5ab71